cargo-dist/src/installer.rs

```diff
@@ -20,20 +21,25 @@ pub(crate) fn generate_install_sh_script(
     info: &InstallerInfo,
 ) -> Result<(), miette::Report> {
     let installer_file = &info.dest_path;
-    let mut file = File::create(installer_file)
+    // Unwrapping here because the path is basically guaranteed to have a parent at this point
+    let dest_dir = installer_file.parent().unwrap();
+    // Potential FIXME: Have an axoasset method that _just_ creates a file, without writing
+    // anything (think `File::create`)
+    LocalAsset::write_new("", installer_file.as_str(), dest_dir.as_str())
```

Comment:
hmm interesting, we ultimately buffer up the entire result before writing to the file anyway (to do newline cleanups), and presumably the more transactional you can make "create a file, write its contents to completion" the better, right? So maybe we should just move the file creation to the end?

Might make it harder to test this code later though...?

Reply:
Also I forget, what's the significance of making us pass the file path and the dir path?

Or wait, I think this is maybe just incorrect code, and you should be getting `installer_file.filename()` or something?

Reply:
It is incorrect, good catch! I'll move the write calls to the end of the function, I think when testing those functions, we might want to structure them somewhat differently anyways, so I don't think that's necessarily a huge blocker.

Reply:
As for why we pass filename and directory name separately, I have no idea, that's what the signature already looked like when I came upon it. We could probably squash those arguments, though.